bpfman/src/cli/args.rs

```diff
@@ -0,0 +1,344 @@
+    /// Required: Program id to be unloaded.
+    pub(crate) id: u32,
+}
+
+#[derive(Args, Debug)]
+pub(crate) struct ListArgs {
+    /// Optional: List a specific program type
+    /// Example: --program-type xdp
+    ///
+    /// [possible values: unspec, socket-filter, kprobe, tc, sched-act,
+    ///                   tracepoint, xdp, perf-event, cgroup-skb,
+    ///                   cgroup-sock, lwt-in, lwt-out, lwt-xmit, sock-ops,
+    ///                   sk-skb, cgroup-device, sk-msg, raw-tracepoint,
+    ///                   cgroup-sock-addr, lwt-seg6-local, lirc-mode2,
+    ///                   sk-reuseport, flow-dissector, cgroup-sysctl,
+    ///                   raw-tracepoint-writable, cgroup-sockopt, tracing,
+    ///                   struct-ops, ext, lsm, sk-lookup, syscall]
+    #[clap(short, long, verbatim_doc_comment, hide_possible_values = true)]
+    pub(crate) program_type: Option<ProgramType>,
+
+    /// Optional: List programs which contain a specific set of metadata labels
+    /// that were applied when the program was loaded with `--metadata` parameter.
+    /// Format: <KEY>=<VALUE>
+    ///
+    /// Example: --metadata-selector owner=acme
+    #[clap(short, long, verbatim_doc_comment, value_parser=parse_key_val, value_delimiter = ',')]
+    pub(crate) metadata_selector: Option<Vec<(String, String)>>,
+
+    /// Optional: List all programs.
+    #[clap(short, long, verbatim_doc_comment)]
+    pub(crate) all: bool,
+}
+
+#[derive(Args, Debug)]
+pub(crate) struct GetArgs {
+    /// Required: Program id to get.
+    pub(crate) id: u32,
+}
+
+#[derive(Subcommand, Debug)]
+pub(crate) enum ImageSubCommand {
+    /// Pull an eBPF bytecode image from a remote registry.
+    Pull(PullBytecodeArgs),
+}
+
+#[derive(Args, Debug)]
+pub(crate) struct PullBytecodeArgs {
+    /// Required: Container Image URL.
+    /// Example: --image-url quay.io/bpfman-bytecode/xdp_pass:latest
+    #[clap(short, long, verbatim_doc_comment)]
+    pub(crate) image_url: String,
+
+    /// Optional: Registry auth for authenticating with the specified image registry.
+    /// This should be base64 encoded from the '<username>:<password>' string just like
+    /// it's stored in the docker/podman host config.
+    /// Example: --registry_auth "YnjrcKw63PhDcQodiU9hYxQ2"
+    #[clap(short, long, verbatim_doc_comment)]
+    pub(crate) registry_auth: Option<String>,
+
+    /// Optional: Pull policy for remote images.
+    ///
+    /// [possible values: Always, IfNotPresent, Never]
+    #[clap(short, long, verbatim_doc_comment, default_value = "IfNotPresent")]
+    pub(crate) pull_policy: String,
+}
+
+#[derive(Args, Debug)]
+pub(crate) struct ServiceArgs {
+    /// Enable CSI support.
+    #[clap(long)]
+    pub(crate) csi_support: bool,
+}
+
+#[derive(Subcommand, Debug)]
+pub(crate) enum SystemSubcommand {
+    /// Load an eBPF program from a local .o file.
+    Service(ServiceArgs),
+}
+
+/// Parse a single key-value pair
```

Comment:
nit these should probably live in a `util.rs`

Reply:
They could, but they are only useed by the functions in this file, so I think it's okay.